src/test/java/de/tum/in/www1/artemis/DatabaseQueryCountTest.java

```diff
@@ -76,4 +86,78 @@ void testGetAllCoursesForDashboardRealisticQueryCount() throws Exception {
         // 2 DB calls to get the quiz batches for active quiz exercises
         // 1 DB call to get all plagiarism cases
     }
+
+    @Test
+    @WithMockUser(username = TEST_PREFIX + "student2", roles = "USER")
+    void testStartWorkingOnExamQueryCount() throws Exception {
+        Exam exam = createActiveExam();
+        StudentExam studentExam = createStudentExamForUser(exam, TEST_PREFIX + "student2");
+
+        assertThatDb(() -> {
+
+            startWorkingOnExam(studentExam);
+            return null;
+
+        }).hasBeenCalledAtMostTimes(getStartWorkingOnExamExpectedTotalQueryCount());
+    }
+
+    @Test
+    @WithMockUser(username = TEST_PREFIX + "student3", roles = "USER")
+    void testSubmitAnswerOfExamQueryCount() throws Exception {
+        Exam exam = createActiveExam();
+        StudentExam studentExam = createStudentExamForUser(exam, TEST_PREFIX + "student3");
+        startWorkingOnExam(studentExam);
+
+        assertThatDb(() -> {
+
+            submitExam(studentExam);
+            return null;
+
+        }).hasBeenCalledAtMostTimes(getSubmitAnswerOfExamExpectedTotalQueryCount());
```

Comment:
You could shorten this either by changing `submitExam` to return the server response or by adding an overloaded `assertThatDb` method taking a Consumer.
```suggestion
        assertThatDb(() -> submitExam(studentExam))
        .hasBeenCalledAtMostTimes(getSubmitAnswerOfExamExpectedTotalQueryCount());
```

Reply:
Changed the submitExam to return the server response which shortened the assertion. Adding an overloaded method with a Consumer argument requires creating a new QueryCountAssertion class as QueryCountAsserttion class requires a ThrowingProducer class.